src/v/kafka/server/handlers/alter_configs.cc

```diff
@@ -337,7 +337,8 @@ create_topic_properties_update(
                   update.properties.leaders_preference,
                   cfg.value,
                   kafka::config_resource_operation::set,
-                  noop_validator<config::leaders_preference>{},
+                  leaders_preference_validator{
+                    ctx.feature_table().local().should_sanction()},
```

Comment:
@BenPope is this what you were saying that it would be good to be avoided or the equivalent in incremental?